notebooks/week2/05.log_and_register_fe_model.py

```diff
@@ -0,0 +1,193 @@
+# Databricks notebook source
+# MAGIC %pip install /Volumes/mlops_dev/hotel_reservation/data/hotel_reservation-0.0.1-py3-none-any.whl
+
+# COMMAND ----------
+
+# MAGIC %restart_python
+
+# COMMAND ----------
+
+
+import matplotlib.pyplot as plt
+import mlflow
+import yaml
+from databricks import feature_engineering
+from databricks.feature_engineering import FeatureLookup
+from mlflow.models import infer_signature
+from pyspark.sql import SparkSession
+from sklearn.compose import ColumnTransformer
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.impute import SimpleImputer
+from sklearn.metrics import ConfusionMatrixDisplay, confusion_matrix, f1_score
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import OneHotEncoder, StandardScaler
+
+# COMMAND ----------
+
+# Initialize the Databricks session and clients
+spark = SparkSession.builder.getOrCreate()
+fe = feature_engineering.FeatureEngineeringClient()
+
+# COMMAND ----------
+
+mlflow.set_registry_uri("databricks-uc")
+mlflow.set_tracking_uri("databricks")
+
+# COMMAND ----------
+
+# Load configuration
+with open("../../project_config.yml", "r") as file:
+    config = yaml.safe_load(file)
+
+# COMMAND ----------
+
+# Extract configuration details
+num_features = config["num_features"]
+cat_features = config["cat_features"]
+target = config["target"]
+parameters = config["parameters"]
+catalog_name = config["catalog_name"]
+schema_name = config["schema_name"]
+id = config["id"]
+
+# COMMAND ----------
+
+# Define table names and function name
+feature_table_name = f"{catalog_name}.{schema_name}.hotel_features"
+
+# COMMAND ----------
+
+# Load training and test sets
+train_set = spark.table(f"{catalog_name}.{schema_name}.train_set")
+test_set = spark.table(f"{catalog_name}.{schema_name}.test_set").toPandas()
+
+# COMMAND ----------
+
+# Create or replace the hotel_features table
+spark.sql(f"""
+CREATE OR REPLACE TABLE {catalog_name}.{schema_name}.hotel_features
+(Booking_ID STRING NOT NULL,
```

Comment:
this can be improved by getting variable names and types from the project config file and using f string